xbmc/cores/VideoPlayer/VideoRenderers/windows/RendererHQ.h

```diff
@@ -25,9 +25,10 @@ class CRendererHQ : public CRendererBase
 
   void SelectPSVideoFilter();
   bool HasHQScaler() const;
+  bool IsNeededScaler() const;
 
   ESCALINGMETHOD m_scalingMethod = VS_SCALINGMETHOD_AUTO;
-  ESCALINGMETHOD m_scalingMethodGui = VS_SCALINGMETHOD_AUTO;
+  ESCALINGMETHOD m_scalingMethodGui = VS_SCALINGMETHOD_MAX;
```

Comment:
`MAX` is just a helper which indicates the end of enum values. you should use it for validation only.

Reply:
I know this but if it is initialized to AUTO and the method is AUTO then the initialization code is never executed nere:

https://github.com/xbmc/xbmc/blob/67b9857799d2d4aff3ad6608ad95c14a5fab7090/xbmc/cores/VideoPlayer/VideoRenderers/windows/RendererHQ.cpp#L94

because (AUTO != AUTO) this condition will never be met

`m_scalingMethodGui` needs to be initialized with "never used value" to force this TRUE first time and execute  `SelectPSVideoFilter();`

Reply:
Are you saying the all others platfoms code are wrong?

e.g.
https://github.com/xbmc/xbmc/blob/67b9857799d2d4aff3ad6608ad95c14a5fab7090/xbmc/cores/VideoPlayer/VideoRenderers/LinuxRendererGL.h#L217

Reply:
And please read

HOW HAS BEEN Tested?

**TEST 2: source 720p --> Display 1080p scale method AUTO**
**Before:**
HQ scalers are NOT enabled since in RendererHQ.h m_scalingMethodGui = VS_SCALINGMETHOD_AUTO; and not triggers initial filter update (works OK if is selected from video OSD menu while playing).
**After:**
HQ scalers are enabled with default convolution kernel 4x4.

Reply:
ok. you are right, it's a best way